main/lsp/LSPTypechecker.cc

```diff
@@ -396,14 +416,14 @@ void LSPTypechecker::commitTypecheckRun(TypecheckRun run) {
     int i = -1;
     ENFORCE(updates.updatedFileIndexes.size() == updates.updatedFileHashes.size() &&
             updates.updatedFileHashes.size() == updates.updatedFiles.size());
+    ENFORCE(globalStateHashes.size() == indexed.size() && globalStateHashes.size() == diagnosticEpochs.size());
     for (auto &ast : updates.updatedFileIndexes) {
         i++;
         const int id = ast.file.id();
         if (id >= indexed.size()) {
             indexed.resize(id + 1);
-        }
-        if (id >= globalStateHashes.size()) {
             globalStateHashes.resize(id + 1);
+            diagnosticEpochs.resize(id + 1);
```

Comment:
Just checking...is `id >= indexed.size()` expected to be a rare occurrence?  (If not, would you want to consider looping before line 420 to calculate the max `id` so that you're guaranteed to resize exactly once?)

Reply:
It should only happen when the user adds a file -- so it's rare. We would only resize multiple times if the user added multiple files at once, which I think would be a very rare occurrence.

I also have a feeling that `resize` is a no-op unless `indexed` is moving up to another size class (typically powers of 2, so 32 to 33 might cause it to resize to 64 internally).